torchrl/envs/libs/openspiel.py

```diff
@@ -0,0 +1,358 @@
+        self.reward_spec = UnboundedContinuousTensorSpec(
+            shape=[*self.batch_size, env.num_players()],
+            device=self.device,
+        )
+        self.done_spec = DiscreteTensorSpec(
+            n=2,
+            shape=torch.Size((1,)),
+            dtype=torch.bool,
+            device=self.device,
+        )
+        action_spec_cls = (
+            DiscreteTensorSpec
+            if self._categorical_action_encoding
+            else OneHotDiscreteTensorSpec
+        )
+        self.action_spec = action_spec_cls(
+            env.num_distinct_actions(),
+            dtype=torch.int64,
+            device=self.device,
+        )
+        self.observation_spec = CompositeSpec(
+            observation=UnboundedContinuousTensorSpec(
+                shape=(
+                    *self.batch_size,
+                    env.num_players(),
+                    *env.get_game().observation_tensor_shape(),
+                ),
+                device=self.device,
+            ),
+            current_player=DiscreteTensorSpec(
+                env.num_players(),
+                dtype=torch.int,
+                device=self.device,
+            ),
+            state=NonTensorSpec([]),
+            shape=self.batch_size,
+        )
+
+    def _set_seed(self, seed):
+        if seed is not None:
+            raise NotImplementedError("This environment has no seed.")
+
+    def current_player(self):
+        return self._env.current_player()
+
+    def _update_action_mask(self):
+        action_mask = torch.zeros(
+            self._env.num_distinct_actions(), device=self.device, dtype=torch.bool
+        )
+        action_mask[self._env.legal_actions()] = True
+        self.action_spec.update_mask(action_mask)
+
+    def _make_td_out(self, exclude_reward=False):
+        observation_shape = self._env.get_game().observation_tensor_shape()
+        observation = torch.stack(
+            [
+                self._to_tensor(self._env.observation_tensor(idx)).reshape(
+                    observation_shape
+                )
+                for idx in range(self._env.num_players())
+            ]
+        )
+        reward = self._to_tensor(self._env.returns())
+        done = torch.tensor(
+            self._env.is_terminal(), device=self.device, dtype=torch.bool
+        )
+        state = self._env.serialize()
+        current_player = torch.tensor(
+            self.current_player(), device=self.device, dtype=torch.int
+        )
+
+        tensordict_out = TensorDict(
+            source={
+                "observation": observation,
+                "done": done,
+                "terminated": done.clone(),
+                "state": state,
+                "current_player": current_player,
+            },
+            batch_size=self.batch_size,
+            device=self.device,
+        )
+
+        if not exclude_reward:
+            tensordict_out["reward"] = reward
```

Comment:
setting it after the fact is more expensive than putting it in the dict or not, maybe we could do 
```python
source = {
                "observation": observation,
                "done": done,
                "terminated": done.clone(),
                "state": state,
                "current_player": current_player,
            }
if not exclude_reward:
    source["reward"] = reward
tensordict_out = TensorDict(source, ...)
```